internal/feature.go

```diff
@@ -61,13 +67,11 @@ func FeatureTest(name, version string, fn FeatureTestFn) func() error {
 
 	ft := new(featureTest)
 	return func() error {
-		ft.Lock()
-		defer ft.Unlock()
-
-		if ft.successful {
+		if ft.isSuccessful() {
 			return ft.result
```

Comment:
Can you move the locking in here? Like

```
ft.RLock()
if ft.successful {
  defer ft.RUnlock()
  return ft.result
}
ft.RUnlock()
```
etc.? Otherwise we need a comment to explain why it's safe to read ft.result without the lock being taken, which is kind of subtle.

Reply:
done.